### PR TITLE
HDDS-13064. [snapshot] Add test coverage for SnapshotUtils.isBlockLocationInfoSame()

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?
HDDS-13064. [snapshot] Add test coverage for SnapshotUtils.isBlockLocationInfoSame()

Please describe your PR in detail:
* Add test cases to cover each and every cases in SnapshotUtils.isBlockLocationInfoSame().
* There is a hsync corner case where the code does not match the comment description. Fixed the code to match the comment.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-13064

## How was this patch tested?

New unit tests added in TestSnapshotUtils.java
https://github.com/jojochuang/ozone/actions/runs/15079727963/job/42394324040

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  private OmKeyLocationInfoGroup createLocationGroup(OmKeyLocationInfo... locs) {
+    OmKeyLocationInfoGroup group = new OmKeyLocationInfoGroup(0, Arrays.asList(locs));
+    return group;
+  }
+
+  private OmKeyInfo createOmKeyInfo(boolean hsync, OmKeyLocationInfoGroup group) {
+    return createOmKeyInfoWithObjectID(hsync, group, 0);
+  }
+
+  private OmKeyInfo createOmKeyInfoWithObjectID(boolean hsync, OmKeyLocationInfoGroup group, long objectId) {
```
Comment (swamirishi) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java:57]: We can may be overload some functions here
https://github.com/apache/ozone/blob/5c5db8e9bc5b8dcf5d586e10aa648ac838700d49/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/OMRequestTestUtils.java#L559

**Review by swamirishi — CHANGES_REQUESTED** [review]
@jojochuang thanks for the patch. LGTM have a nitpicky comment on reusing functions already present in util functions

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  private OmKeyInfo createOmKeyInfo(boolean hsync, OmKeyLocationInfoGroup group) {
+    return createOmKeyInfo(hsync, group, 0);
+  }
+
+  private OmKeyInfo createOmKeyInfo(boolean hsync, OmKeyLocationInfoGroup group, long objectId) {
+    return createOmKeyInfo(hsync, Collections.singletonList(group), objectId);
+  }
+
+  private OmKeyInfo createOmKeyInfo(boolean hsync, List<OmKeyLocationInfoGroup> group, long objectId) {
+    OmKeyInfo keyInfo = new OmKeyInfo.Builder()
```
Comment (swamirishi) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java:63]: nit: I was just wondering if you can have a similar function in https://github.com/apache/ozone/blob/5c5db8e9bc5b8dcf5d586e10aa648ac838700d49/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/OMRequestTestUtils.java#L559-L560 and call it here. Most of function parameters are the same.

**Review by swamirishi — APPROVED** [review]
LGTM

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java (hunk 2 above)
Comment (jojochuang) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java:63]: I'll look into this later as a separate task. We have a lot of places that do similar work, so might as well to address that holistically.

**Comment by jojochuang** [comment]
Merging the PR based on @swamirishi s review. Thanks!
